packages/flutter_tools/lib/src/devtools_launcher.dart

```diff
@@ -97,9 +97,9 @@ class DevtoolsServerLauncher extends DevtoolsLauncher {
       bool devToolsActive = await _checkForActiveDevTools();
       if (!offline) {
         await _activateDevTools(throttleUpdates: devToolsActive);
-      }
-      if (!devToolsActive && !offline) {
-        devToolsActive = await _checkForActiveDevTools();
+        if (!devToolsActive) {
```

Comment:
why this change? if we know that we can't reach pub, then re-activating devtools will almost always fail - unless it is already activated. but then `_checkForActiveDevTools` should have already come back as true

Reply:
This is a no-op change. It's semantically identical, except in principle might save a couple of cycles on a redundant check if the compiler didn't notice that it was semantically identical.